### PR TITLE
envoy-v3-bootstrap-config

### PR DESCRIPTION
**Pull request by arianaw66**

### CODE REVIEW

**Comment by arianaw66** [comment]
depends on https://github.com/solo-io/envoy-operator/pull/12

File: install/helm/gloo/templates/9-gateway-proxy-configmap.yaml [hunk 1]
```diff
@@ -124,8 +127,8 @@ data:
                               route:
                                 cluster: admin_port_cluster
                     http_filters:
-                      - name: envoy.router
-                        config: {}
+                      - name: envoy.filters.http.router
+                        typed_config: {}
```
Comment (EItanya) [inline install/helm/gloo/templates/9-gateway-proxy-configmap.yaml:131]: Can remove this line

**Comment by arianaw66** [comment]
depends on https://github.com/solo-io/envoy-operator/pull/13

File: install/nomad/jobs/gloo.nomad [hunk 2]
```diff
@@ -343,8 +344,7 @@ static_resources:
                             prefix_rewrite: "/stats/prometheus"
                             cluster: admin_port_cluster
                 http_filters:
-                  - name: envoy.router
-                    config: {}
+                  - name: envoy.filters.http.router
```
Comment (kdorosh) [inline install/nomad/jobs/gloo.nomad:348]: do we still need to provide an empty config here?

File: install/test/fixtures.go [hunk 3]
```diff
@@ -99,11 +99,11 @@ static_resources:
...
     - filters:
-      - config:
-          codec_type: auto
+      - typed_config:
+          "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
+          codec_type: AUTO
           http_filters:
-          - config: {}
-            name: envoy.router
+          - name: envoy.filters.http.router
```
Comment (kdorosh) [inline install/test/fixtures.go:106]: i guess it appears we don't

File: install/test/fixtures.go [hunk 4]
```diff
@@ -419,20 +421,21 @@ static_resources:
                   cluster: admin_port_cluster
                   prefix_rewrite: /stats/prometheus
           stat_prefix: prometheus
-        name: envoy.http_connection_manager
+          tracing:
+            provider:
+              typed_config:
+                '@type': type.googleapis.com/envoy.config.trace.v2.ZipkinConfig
+                collector_cluster: zipkin
+                collector_endpoint: /api/v1/spans
```
Comment (kdorosh) [inline install/test/fixtures.go:429]: i think this should be v2
```suggestion
                collector_endpoint: /api/v2/spans
```

File: test/e2e/cors_test.go [hunk 5]
```diff
@@ -45,7 +45,7 @@ var _ = Describe("CORS", func() {
 	var td corsTestData
 
 	const (
-		corsFilterString       = `"name": "envoy.cors"`
+		corsFilterString       = `"name": "envoy.filters.http.cors"`
```
Comment (kdorosh) [inline test/e2e/cors_test.go:48]: we should be able to remove this and use `wellknown` from envoy go-control-plane

File: install/nomad/jobs/gloo.nomad (hunk 2 above)
Comment (kdorosh) [reply install/nomad/jobs/gloo.nomad:348]: ah i see eitan's comment above, we don't need https://github.com/solo-io/gloo/pull/3343#discussion_r455331191

File: install/test/fixtures.go (hunk 3 above)
Comment (kdorosh) [reply install/test/fixtures.go:106]: https://github.com/solo-io/gloo/pull/3343#discussion_r455331191

File: changelog/v1.5.0-beta9/upgrade-envoy-bootstrap-config-to-v3.yaml [hunk 6]
```diff
@@ -0,0 +1,5 @@
+changelog:
+- type: NON_USER_FACING
+  issueLink: https://github.com/solo-io/gloo/issues/2815
+  resolvesIssue: false 
+  description: This is part of the upgrade to v3 of the Envoy xDS API. It upgrades the bootrstrap config from v2 to v3.
```
Comment (kdorosh) [inline changelog/v1.5.0-beta9/upgrade-envoy-bootstrap-config-to-v3.yaml:5]: typo,
```suggestion
  description: This is part of the upgrade to v3 of the Envoy xDS API. It upgrades the bootstrap config from v2 to v3.
```

**Pull request merged**